frontend/src/views/SettingsView.vue

```diff
@@ -1,7 +1,291 @@
+              <span class="interactive-text">Add/remove facilities</span>
+            </td>
+          </tr>
+          </template>
+          </v-data-table>
+        </v-skeleton-loader>
+      </v-col>
+    </v-row>
+  </v-container>
 </template>
 
-<script></script>
+<script>
+import { mapActions, mapState } from 'pinia'
+import { useAuthStore } from '@/stores/auth'
+import { useUsersStore } from '@/stores/users'
+import rolesMixin from '@/mixins/rolesMixin.js'
+import { CRM_STATE_CODES } from '@/utils/constants'
+
+export default {
+  mixins: [rolesMixin],
+  data() {
+    return {
+      showFilterInput: false,
+      facilityNameFilter: '',
+      expanded: [],
+      headersUsers: [
+        { title: '', key: 'data-table-expand', width: '6%' },
+        { title: 'First Name', key: 'firstName', width: '10%' },
+        { title: 'Last Name', key: 'lastName', width: '10%' },
+        { title: 'Email', key: 'email', width: '12%' },
+        { title: 'Username', key: 'userName', width: '12%' },
+        { title: 'Role', key: 'roles', width: '12%' },
+        { title: 'Expense Authority', key: 'isExpenseAuthority', width: '12%' },
+        { title: 'Status', key: 'stateCode', width: '12%' },
+      ],
+      headersFacilities: [
+        { title: 'Facility Name', key: 'name', width: '40%' },
+        { title: 'Address', key: 'address', width: '60%' },
+      ],
+    }
+  },
+  computed: {
+    ...mapState(useAuthStore, ['userInfo']),
+    ...mapState(useUsersStore, ['usersAndFacilities']),
+
+    filteredUserFacilities() {
+      return this.usersAndFacilities.filter(user =>
+        user.facilities.some(facility => facility.name.toLowerCase().includes(this.facilityNameFilter.toLocaleLowerCase())))
+    },
+  },
+  async created() {
+    this.getUsersFacilities(this.organizationId)
+  },
+  methods: {
+    ...mapActions(useUsersStore, ['getUsersPermissionsFacilities']),
+
+    /**
+     * Get the users, permissions, and facilities for an organization
+     */
+    async getUsersFacilities(organizationId) {
+      await this.getUsersPermissionsFacilities(this.userInfo.organizationId)
+    },
+
+    /**
+     * Toggle the facility input filter
+     */
+    toggleShowFilter() {
+      if (this.showFilterInput) {
+        this.showFilterInput = false
+        this.facilityNameFilter = ''
+      } else {
+        this.showFilterInput = true
+      }
+    },
+
+    /**
+     * Toggle expanded row
+     */
+    toggleExpand(item) {
+      const index = this.expanded.indexOf(item.contactId);
+      if (index > -1) {
+        // Collapse if already expanded
+        this.expanded.splice(index, 1);
+      } else {
+        // Expand the clicked row
+        this.expanded = [item.contactId];
+      }
+    },
+
+    // TODO: Hard coding of role names in this method is temporary. Will be replaced with endpoint call to get role names.
+    /**
+     * Get the role description for role code(s)
+     */
+    getRoleDescriptions(roles) {
```

Comment:
This method kinda does two things that should be split up:
- Converts roles array to a single comma-delimited String
- Maps role codes to descriptions
If you split out the latter then it'll be an easier refactor later.

Reply:
Fixed.

Reply:
Fixed.